src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion.py

```diff
@@ -114,7 +116,7 @@ def __init__(
             new_config["clip_sample"] = False
             scheduler._internal_dict = FrozenDict(new_config)
 
-        if safety_checker is None:
+        if safety_checker is None and requires_safety_checker:
```

Comment:
I don't think this is robust enough, or I'm not fully understanding. Consider the following case:

- `requires_safety_checker` is `False`
- The pre-trained repo does contain a safety checker model.
- The user passes `safety_checker=None` to `from_pretrained`

Wouldn't the warning still be displayed by `DiffusionPipeline.from_pretrained`?  https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipeline_utils.py#L573

Reply:
Hmm no if `requires_safety_checker` is False then the warning cannot be displayed - can you add a repro code snippet?

Reply:
```Python
pipeline = DiffusionPipeline.from_pretrained("diffusers/tiny-torch-full-checker", safety_checker=None)
```

It's like the tiny model but with a safety checker (sorry I uploaded a full-sized one). Also, `requires_safety_checker` is `False` in the config.

Reply:
Your command above doesn't produce a warning as it shouldn't